docs/development.txt

```diff
@@ -0,0 +1,82 @@
+* Internal shared data-structure or API change
+
+For these sorts of changes, contributors will:
+
+* Create one or more issues in the Github issue tracker
+* Start a [DISCUSS] thread on the https://groups.google.com/forum/#!forum/janusgraph-dev[janusgraph-dev] list where the proposed change may be discussed by committers
+and other community members
+* When the proposer feels it appropriate, a VOTE shall be called. As with pull requests three +1 votes, including
+one from the relevant Technical Steering Committee (TSC) member if the change is limited to a single project, are required for the change to be accepted.
+
+Though not required, it is strongly recommended that contributors limit the amount of development work they
+perform before their change is accepted. In the instance where the change is not approved or significant updates to
+the proposal are made, we wish to minimize the amount of rework required.
+
+[[branching]]
+Branching
+---------
+For features that involve only one developer, developers will work in their own JanusGraph forks, managing their
+own branches, and submitting pull requests when ready. Feature branches in the main JanusGraph repository will be
+created when more than one developer collaborates on a feature.
+
+Branch Naming Conventions
+~~~~~~~~~~~~~~~~~~~~~~~~~
+All branch names shall be prepended with `Issue_#_`.
+
+[[pull-requests]]
+Pull Requests
+-------------
+
+Most changes to JanusGraph must be reviewed before they are committed (RTC). This workflow is performed using Github's
+pull request process. In instances where the committer deems the full process
+unnecessary, commit-then-review (CTR) may be employed. This should be used infrequently and only for extremely trivial
+changes. Changes of this sort may include:
+
+* Documentation typo or small documentation addition
+* Addition of a new test case
+
+Any commit that is made following CTR shall include the fact that it is a CTR in the commit comments.
+
+Users wishing to contribute to JanusGraph should fork the https://github.com/janusgraph/janusgraph[JanusGraph]
+repository and then submit pull requests using the Github pull request process. Please include the relevant issue
+number in the title of your pull request. A vote (-1, 0, +1) is required before a pull request may be merged.
+Non-committers are free, and encouraged to vote. Their votes will be non-binding, but can be taken into consideration
+and are still very valuable community input. The following rules apply to the voting process.
+
+* Three +1 votes are required merge a pull request
+* One or more -1 votes will veto the pull request until the noted issue(s) are addressed and the pull request
+is resubmitted
+* Committers may vote on their own pull requests
+* Pull requests that are limited to one component require at least one of the +1 votes to come from the relevant
+TSC member. If TSC members are unable to review after 2 weeks, the pull request may be merged
+after three +1 votes
```

Comment:
I'll remove the TSC requirement.  I think that's too stringent to start and we always add it in later if need be.